docs/source/bufr4.rst

```diff
@@ -0,0 +1,46 @@
+.. _bufr4:
+
+Anatomy of a BUFR4 message
+--------------------------
+
+.. role:: redtext
+
+In order to understand the mapping between a CSV file and its BUFR encoding it is first helpful to understand
+the anatomy of a BUFR message.
+A BUFR message is encoded in binary and contains 6 sections as shown in the diagram below.
+The ecCodes keys for the different elements are also shown as these are used to set certain elements
+(highlighted in red) as part of the conversion to BUFR.
+The non-highlighted elements (including those without an ecCodes key) are either set by the eccodes module based on the data,
+have a default value or can be omitted / set to missing.
+
+The information cotained in Sections 0, 1, and 3 are essentially metadata specifying:
+
+* the version of the BUFR tables used (:redtext:`editionNumber, masterTableNumber, masterTableVersionNumber`);
+* where the data have come from (:redtext:`bufrHeaderCentre, bufrHeaderSubCentre`);
+* the typical time of the observation (:redtext:`typicalYear ... typicalSecond`);
+* the type of data and what parameters are included (:redtext:`dataCategory, internationalDataSubCategory and unexpandedDescriptors`).
+
+These are all specified in the BUFR template mapping file.
+
+The :redtext:`edition number` and :redtext:`master table number` should be 4 and 0 respectively.
+BUFR edition 4 is the latest version and whilst it is possible to define new BUFR tables for different application
+areas only Master Table 0 (MT0) has been defined (meteorology).
+The tables in MT0 are updated approximately every 6 months as part of the WMO fast track process.
+The latest version can be found at https://github.com/wmo-im/BUFR4 and it is recommended to use the the most recent
+version for the :redtext:`master table version number`.
```

Comment:
Clarify that tags in this repo represent latest stable version (whereas master is working/tip).

Reply:
done